src/d/actor/d_a_kui.cpp

```diff
@@ -4,68 +4,541 @@
+        J3DMaterial* mat = modelData->getMaterialNodePointer(i);
+        for (u32 j = 0; j < 8; j++) {
+            J3DTexMtx* texMtx = mat->getTexMtx(j);
+            if (texMtx != NULL) {
+                switch (texMtx->getTexMtxInfo().mInfo) {
+                    case 10:
+                    case 11:
+                        texMtx->getTexMtxInfo().setEffectMtx(now_copy);
+                        break;
+                }
+            }
+        }
+    }
 }
 
 /* 0000037C-00000540       .text daKui_Draw__FP9kui_class */
-static BOOL daKui_Draw(kui_class*) {
-    /* Nonmatching */
+static BOOL daKui_Draw(kui_class* i_this) {
+    if (i_this->type == 3) {
+        u32 light_type = 0;
+        if (REG0_S(0) != 0) {
+            light_type = 1;
+        }
+
+        i_this->eyePos = i_this->current.pos;
+        i_this->eyePos.y += REG0_F(10);
+
+        g_env_light.settingTevStruct(light_type, &i_this->eyePos, &i_this->tevStr);
+        g_env_light.setLightTevColorType(i_this->mpModel2, &i_this->tevStr);
+        g_env_light.setLightTevColorType(i_this->mpModel, &i_this->tevStr);
+
+        setEffectMtx(i_this, i_this->mpModel2->getModelData(), REG0_F(11) + 1.0f);
+        setEffectMtx(i_this, i_this->mpModel->getModelData(), REG0_F(12) + 1.0f);
+
+        mDoExt_modelUpdateDL(i_this->mpModel2);
+        mDoExt_modelUpdateDL(i_this->mpModel);
+    } else if (i_this->type != 1) {
+        g_env_light.settingTevStruct(TEV_TYPE_ACTOR, &i_this->current.pos, &i_this->tevStr);
+        g_env_light.setLightTevColorType(i_this->mpModel2, &i_this->tevStr);
+
+        dComIfGd_setListBG();
+        mDoExt_modelUpdateDL(i_this->mpModel2);
+        dComIfGd_setList();
+
+        if (i_this->mpModel) {
+            g_env_light.setLightTevColorType(i_this->mpModel, &i_this->tevStr);
+            mDoExt_modelUpdateDL(i_this->mpModel);
+        }
+    }
+    return TRUE;
 }
 
 /* 00000540-00000920       .text demo_camera__FP9kui_class */
-void demo_camera(kui_class*) {
-    /* Nonmatching */
+static void demo_camera(kui_class* i_this) {
+    camera_class* camera = dComIfGp_getCamera(dComIfGp_getPlayerCameraID(0));
+    s8 bVar2 = true;
+
+    switch ((s8)i_this->field_0x2E8) {
+        case 0:
+            break;
+        case 1:
+            if (!i_this->eventInfo.checkCommandDemoAccrpt()) {
+                fopAcM_orderPotentialEvent(i_this, dEvtFlag_STAFF_ALL_e, 0xFFFF, 0);
+                i_this->eventInfo.onCondition(dEvtCnd_UNK2_e);
+                bVar2 = false;
+                break;
+            }
+            i_this->field_0x2E8 = i_this->field_0x2E8 + 1;
+            camera->mCamera.Stop();
+            camera->mCamera.SetTrimSize(2);
+            i_this->field_0x2EA = 0;
+            // fallthrough
+        case 2:
+            i_this->current.pos.y = i_this->home.pos.y + i_this->field_0x30C;
+
+            s32 uVar3 = (s32) i_this->field_0x2EA;
+            if (uVar3 < 20) {
+                f32 sin_result = cM_ssin((uVar3 & 0x1F) << 11);
```

Comment:
Probably intended to be a multiplication since it's an angle
```suggestion
                f32 sin_result = cM_ssin((uVar3 & 0x1F) * 0x800);
```